src/main/java/org/semux/api/util/TransactionBuilder.java

```diff
@@ -72,6 +72,9 @@
      */
     private byte[] data;
 
+    private long gasPrice = 0;
```

Comment:
Maybe it would be better to have `GasPrice` and `GasAmount` classes, so type system could help with some potential nasty bugs? This is why `Amount` was introduced. Having `long` this, `long` that makes more room for mistakes.

Reply:
its a reasonable point, but I'm not sure we need holder classes for them.

How about utility methods to Amount for calculating Amount from (gas,gasPrice), but I think thats about it.

I think only thing we need to worry about is someone trying to overflow/underflow the long,